src/main/java/com/vmware/vipclient/i18n/VIPCfgFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class VIPCfgFactory {
+
+    /**
+     * A lazy-loaded singleton VIPCfg using the "initialization-on-demand holder" design pattern
+     */
+    private static class GlobalCfgHolder {
+        static final VIPCfg globalCfg = new VIPCfg();
+    }
+
+    public static VIPCfg getGlobalCfg() {
+        return GlobalCfgHolder.globalCfg;
+    }
+
+    /**
+     * A lazy-loaded singleton Map<String, VIPCfg> configs using the "initialization-on-demand holder" design pattern
+     */
+    private static class ConfigsHolder {
+        static final Map<String, VIPCfg> configs = new ConcurrentHashMap<>();
```

Comment:
Because config name may be changed separately, there should be a method to deal with this to change the key.

Reply:
I have deprecated the capability to instantiate VIPCfg with no product name. See new code changes.